src/components/templates/product-availability-modal/sales-channel-availability-table.tsx

```diff
@@ -0,0 +1,226 @@
+import { SalesChannel } from "@medusajs/medusa"
+import React, { useMemo } from "react"
+import { TableInstance } from "react-table"
+import Button from "../../fundamentals/button"
+import PlusIcon from "../../fundamentals/icons/plus-icon"
+import IndeterminateCheckbox from "../../molecules/indeterminate-checkbox"
+import { LayeredModalContext } from "../../molecules/modal/layered-modal"
+import Table, { TablePagination } from "../../molecules/table"
+import { useAddChannelsModalScreen } from "./add-channels-modal-screen"
+
+type SalesChannelAvailabilityTableProps = {
+  salesChannels: SalesChannel[]
+  limit: number
+  offset: number
+  setOffset: (offset: number) => void
+  setQuery: (query: string) => void
+  setSelectedRowIds?: (selectedRowIds: string[]) => void
+  tableAction?: React.ReactNode
+  tableState: TableInstance<SalesChannel>
+}
+
+type SalesChannelTableActionsProps = {
+  selectedRowIds: string[]
+  chosenSalesChannelIds: string[]
+  onAddSalesChannelsToSelected: (salesChannels: SalesChannel[]) => void
+  onDeselect: () => void
+  onRemove: () => void
+}
+
+export const useAvailableChannelsModalTableColumns = () => {
+  const columns = useMemo(
```

Comment:
suggestion: shouldn't we move the `useMemo` above to wrap it entirely and avoiding complete re render? or does it do the same at this level?